fields/types/gcsavatar/GcsAvatarType.js

```diff
@@ -0,0 +1,378 @@
+
+  options.nofilter = true;
+
+  gcsavatar.super_.call(this, list, path, options);
+
+  // validate gcs config (has to happen after super_.call)
+  if (!keystone.get('gcs config')) {
+    throw new Error('Invalid Configuration\n\n' + 'gcsavatar fields (' + list.key + '.' + path + ') require the "gcs config" option to be set.\n\n' + 'See http://keystonejs.com/docs/configuration/#services-gcsavatar for more information.\n');
+  }
+}
+
+/*!
+ * Inherit from Field
+ */
+
+util.inherits(gcsavatar, super_);
+
+/**
+ * Exposes the custom or keystone gcs config settings
+ */
+
+Object.defineProperty(gcsavatar.prototype, 'gcsConfig', {
+  get: function() {
+    return this.options.gcsConfig || keystone.get('gcs config');
+  },
+});
+
+/**
+ * Registers the field on the List's Mongoose Schema.
+ *
+ * @api public
+ */
+
+gcsavatar.prototype.addToSchema = function() {
+
+  var _this = this;
+  var schema = this.list.schema;
+
+  var paths = this.paths = {
+    // fields
+    filename: this._path.append('.filename'),
+    filetype: this._path.append('.filetype'),
+    gcsBucket: this._path.append('.gcsBucket'),
+    gcsDir: this._path.append('.gcsDir'),
+
+    // virtuals
+    action: this._path.append('_action'),
+    exists: this._path.append('.exists'),
+    upload: this._path.append('_upload'),
+  };
+
+  var schemaPaths = this._path.addTo({}, {
+    filename: String,
+    filetype: String,
+    gcsBucket: String,
+    gcsDir: String,
+  });
+
+  schema.add(schemaPaths);
+
+  var exists = function(item) {
+    return (item.get(paths.filename) ? true : false);
+  };
+
+  // The .exists virtual indicates whether a file is stored
+  schema.virtual(paths.exists).get(function() {
+    return schemaMethods.exists.apply(this);
+  });
+
+  var reset = function(item) {
+    item.set(_this.path, {
+      filename: '',
+      filetype: '',
+      gcsBucket: '',
+      gcsDir: '',
+    });
+  };
+
+  var schemaMethods = {
+    exists: function() {
+      return exists(this);
+    },
+    /**
+     * Resets the value of the field
+     *
+     * @api public
+     */
+    reset: function() {
+      reset(this);
+    },
+    /**
+     * Deletes the file from gcs and resets the field
+     *
+     * @api public
+     */
+    delete: function() {
+      var _this = this;
+      var promise = new Promise(function(resolve, reject) {
+        var gcsConfig = _this.gcsConfig;
+        var bucket = gcsHelper.initBucket(gcsConfig, _this.get(paths.gcsBucket));
+        var filename = _this.get(paths.filename);
+        if (filename && typeof filename === 'string') {
+          var filenameWithoutExt = filename.split('.')[0];
+          bucket.deleteFiles({
+            prefix: _this.get(paths.path) + filenameWithoutExt,
+          }, function(err) {
+            if (err) {
+              return reject(err);
+            }
+            resolve();
+          });
+        } else {
+          resolve();
+        }
+      });
+      reset(this);
+      return promise;
```

Comment:
would we want to provide the avatar deletion and reset? This would also involve the cookie reset.

Reply:
Updated in https://github.com/twreporter/keystone/pull/202/commits/8fa5a7beef095ae84a724b4742f77c95884f7cf4 and https://github.com/twreporter/plate/pull/263/commits/42fabaa8905aeb8bd7f33cae31a77a403dd5cd73